jvm/src/main/java/com/microsoft/bond/protocol/FastBinaryReader.java

```diff
@@ -0,0 +1,147 @@
+package com.microsoft.bond.protocol;
+
+import com.microsoft.bond.BondDataType;
+
+import java.io.InputStream;
+
+public class FastBinaryReader<S extends InputStream> implements TaggedProtocolReader {
+    public final S stream;
+    private final short version;
+    private final BinaryStreamReader<S> reader;
+
+    public FastBinaryReader(final S stream, final short version) {
+        this.stream = stream;
+        this.version = version;
+        this.reader = new BinaryStreamReader<S>(stream);
+    }
+
+    private BondDataType readType() {
+        return new BondDataType(readInt8());
```

Comment:
The behavior I was counting on was ending up with a `BondDataType` that doesn't compare with any of the defined values. The cpp fast binary header does mandate that the top three bits are 0, but values 19-31 would still depend on this behavior.

I'm inclined to leave it.